crates/runtime/tests/models/models_http_endpoint.rs

```diff
@@ -0,0 +1,348 @@
+                Value::String("disabled".to_string()),
+            );
+
+            let app = AppBuilder::new("model_endpoint")
+                .with_model(responses_model)
+                .with_model(chat_model)
+                .build();
+
+            let api_config = create_api_bindings_config();
+            let http_base_url = format!("http://{}", api_config.http_bind_address);
+            let rt = Arc::new(Runtime::builder().with_app(app).build().await);
+
+            let rt_ref_copy = Arc::clone(&rt);
+            tokio::spawn(async move {
+                Box::pin(rt_ref_copy.start_servers(api_config, None, EndpointAuth::no_auth())).await
+            });
+
+            tokio::select! {
+                () = tokio::time::sleep(std::time::Duration::from_secs(60)) => {
+                    return Err(anyhow::anyhow!("Timed out waiting for components to load"));
+                }
+                () = Arc::clone(&rt).load_components() => {}
+            }
+
+            runtime_ready_check(&rt).await;
+
+            let models_url = format!("{http_base_url}/v1/models");
+            let client = Client::new();
+            let response = client
+                .get(&models_url)
+                .send()
+                .await
+                .map_err(anyhow::Error::from)?;
+
+            assert!(
+                response.status().is_success(),
+                "Expected 200 OK, got {}",
+                response.status()
+            );
+
+            let models_response: OpenAIModelResponse =
+                response.json().await.map_err(anyhow::Error::from)?;
+
+            assert_eq!(models_response.object, "list");
+            assert!(
+                !models_response.data.is_empty(),
+                "Expected at least one model in response"
+            );
+
+            let model_names: Vec<&str> =
+                models_response.data.iter().map(|m| m.id.as_str()).collect();
+            assert!(
+                model_names.contains(&"chat_model"),
+                "Expected chat_model in response"
+            );
+            assert!(
+                model_names.contains(&"responses_model"),
+                "Expected responses_model in response"
+            );
+
+            for model in &models_response.data {
+                assert_eq!(model.object, "model");
+                assert_eq!(model.owned_by, "openai:gpt-4o-mini");
+
+                // Check metadata is not included when not requested
+                assert!(
+                    model.metadata.is_none(),
+                    "Expected no metadata when not requested"
+                );
+                assert!(
+                    model.status.is_none(),
+                    "Expected no status when not requested"
+                );
+            }
+            Ok(())
+        })
+        .await
+}
+
+#[tokio::test]
+async fn test_models_http_endpoint_status_no_metadata() -> Result<(), anyhow::Error> {
+    let _tracing = init_tracing(None);
```

Comment:
Because spinning up `spiced` instances is the longest time in some of these tests, we prefer running many tests cases for a given configuration. For example, in our search tests 
https://github.com/spiceai/spiceai/blob/716493c792dedc3ff1433896e9754688f137922b/crates/runtime/tests/models/s3_vectors.rs#L57

Then, as a mental model, a single snapshot is often a sub-test.